src/Route.php

```
@@ -277,34 +310,31 @@ public function __toString(): string
         return $result;
     }
 
-    public function getName(): string
-    {
-        return $this->name ?? (implode(', ', $this->methods) . ' ' . $this->host . $this->pattern);
-    }
-
-    public function getMethods(): array
-    {
-        return $this->methods;
-    }
-
-    public function getPattern(): string
-    {
-        return $this->pattern;
-    }
-
-    public function getHost(): ?string
+    /**
+     * @param string $key
+     *
+     * @return mixed
+     */
+    public function getData(string $key)
     {
-        return $this->host;
-    }
+        $dataMap = $this->getDataMap();
 
-    public function isOverride(): bool
-    {
-        return $this->override;
+        return $dataMap[$key] ?? null;
```

It is better to throw an exception instead of returning null in this case.